modules/programs/git.nix

```diff
@@ -447,6 +465,19 @@ in {
         gpg.program = cfg.signing.gpgPath;
       };
     })
+    (mkIf (cfg.hooks != { }) {
+      programs.git.iniContent = {
+        core.hooksPath = "${pkgs.symlinkJoin {
+          name = "git-hooks";
+          paths = attrValues (mapAttrs (name: text:
+            (pkgs.writeTextFile {
```

Comment:
As is, this requires the user to declare a shell interpreter. We _could_ automatically prepend that here instead.

Reply:
Why not instead use `types.path` and then allow users to choose how they want to generate the scripts? For example, they could do:

```
pre-commit = writeShellScript "cool-script" '...';
```

Reply:
Hmmmm... that would certainly be more flexible. It doesn't feel as clean to me though. @rycee thoughts?

Reply:
I think having it expect the path to a script seems quite reasonable. Then it would also be more straight-forward to pull scripts from external sources. For example if there is a company specified pre-commit hook then you could define a package for it and do `hooks.pre-commit = pkgs.acme-corp.pre-commit-hook`.

I wonder, though if one would like to handle the situation where you have multiple hook scripts? I.e., should it be possible to do something like

``` nix
hooks.pre-commit = mkMerge [
  (writeShellScript "check-trailing-whitespace" "…")
  (writeShellScript "check-no-deploy-credentials" "…")
]
```

?